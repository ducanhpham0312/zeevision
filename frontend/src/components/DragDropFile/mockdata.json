{
  "file": {
    "content": "Test File DragnDrop",
    "name": "test.bpmn",
    "type": "text/plain"
  }
}
