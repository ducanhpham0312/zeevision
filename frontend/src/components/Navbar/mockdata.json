[
  {
    "name": "PROCESSES",
    "path": "/processes"
  },
  {
    "name": "INSTANCES",
    "path": "/instances"
  },
  {
    "name": "INCIDENTS",
    "path": "/incidents"
  },
  {
    "name": "JOBS",
    "path": "/jobs"
  },
  {
    "name": "MESSAGES",
    "path": "/messages"
  },
  {
    "name": "ERRORS",
    "path": "/errors"
  }
]
