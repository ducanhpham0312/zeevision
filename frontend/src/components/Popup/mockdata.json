{
  "openPopup": {
    "open": true,
    "title": "Test Title",
    "children": ["Test child 1", "Test child 2"]
  },
  "closedPopup": {
    "open": false,
    "title": "Test Title",
    "children": ["Test child 1", "Test child 2"]
  },
  "primary": {
    "open": true,
    "title": "Deploy a Process",
    "children": []
  },
  "scrollable": {
    "open": true,
    "title": "Deploy a Process",
    "children": [
      "Cras mattis consectetur purus sit amet fermentum.",
      "Cras justo odio, dapibus ac facilisis in, egestas eget quam.",
      "Morbi leo risus, porta ac consectetur ac, vestibulum at eros.",
      "Praesent commodo cursus magna, vel scelerisque nisl consectetur et."
    ]
  }
}
