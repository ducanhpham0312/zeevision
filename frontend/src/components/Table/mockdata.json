{
  "instances": {
    "orientation": "vertical",
    "alignment": "left",
    "headers": [
      "Instance key",
      "BPMN process ID",
      "Process key",
      "State",
      "Time"
    ],
    "content": [
      [
        2251799813984108,
        "money-loan",
        2251799813685249,
        "ELEMENT_ACTIVATING",
        1696667765
      ],
      [
        2251799813984123,
        "multi-instance-process",
        2251799813685123,
        "ELEMENT_ACTIVATED",
        1696667123
      ],
      [
        2251799813984456,
        "multi-instance-process",
        2251799813685234,
        "ELEMENT_ACTIVATED",
        1696667543
      ],
      [
        2251799813984678,
        "multi-instance-process",
        2251799813685456,
        "ELEMENT_ACTIVATED",
        1696667753
      ],
      [
        2251799813984789,
        "multi-instance-process",
        2251799813685789,
        "ELEMENT_ACTIVATED",
        1696667263
      ]
    ]
  },
  "vertical": {
    "orientation": "vertical",
    "alignment": "left",
    "headers": [
      "Instance key",
      "BPMN process ID",
      "Version",
      "Process key",
      "State",
      "Start time",
      "End time"
    ],
    "content": [
      [
        2251799813984108,
        "multi-instance-process",
        1,
        2251799813685249,
        "ELEMENT_ACTIVATING",
        1696667765,
        1696667800
      ]
    ]
  },
  "horizontal": {
    "orientation": "horizontal",
    "alignment": "center",
    "headers": [
      "Element ID",
      "Variable name",
      "Variable Value",
      "Flow Scope Key",
      "Time created",
      "Option"
    ],
    "content": [
      [
        "multi-instance-process",
        "items",
        "[{\"id\":1,\"value\":10,\"inCart\":10,\"inStock\":20},{\"id\":2,\"value\":20,\"inCart\":2,\"inStock\":10}]",
        2251799813984108,
        1696667765,
        "Show History"
      ],
      [
        "multi-instance-process",
        "isValid",
        "true",
        2251799813984108,
        1696667765,
        "Show History"
      ],
      [
        "multi-instance-process",
        "total",
        "200",
        2251799813984108,
        1696667765,
        "Show History"
      ],
      [
        "another-multi-instance-process",
        "anotherVariable",
        "string",
        2251799813984432,
        1696667333,
        "Show History"
      ],
      [
        "new-multi-instance-process",
        "kmean",
        "newValue",
        2251799813984100,
        1696667542,
        "Show History"
      ],
      [
        "sixth-multi-instance-process",
        "amp",
        "integer",
        2251799813984642,
        1696667164,
        "Show History"
      ],
      [
        "multi-instance-process",
        "items",
        "[{\"id\":1,\"value\":10,\"inCart\":10,\"inStock\":20},{\"id\":2,\"value\":20,\"inCart\":2,\"inStock\":10}]",
        2251799813984108,
        1696667765,
        "Show History"
      ],
      [
        "multi-instance-process",
        "isValid",
        "true",
        2251799813984108,
        1696667765,
        "Show History"
      ],
      [
        "multi-instance-process",
        "total",
        "200",
        2251799813984108,
        1696667765,
        "Show History"
      ],
      [
        "another-multi-instance-process",
        "anotherVariable",
        "string",
        2251799813984432,
        1696667333,
        "Show History"
      ],
      [
        "new-multi-instance-process",
        "kmean",
        "newValue",
        2251799813984100,
        1696667542,
        "Show History"
      ],
      [
        "sixth-multi-instance-process",
        "amp",
        "integer",
        2251799813984642,
        1696667164,
        "Show History"
      ]
    ]
  }
}
