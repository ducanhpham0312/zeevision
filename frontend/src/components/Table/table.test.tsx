import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Table } from ".";
import * as mockdata from "./mockdata.json";

const { headers: verticalHeaders, content: verticalContent } =
  mockdata.vertical;

const { headers: horizontalHeaders, content: horizontalContent } =
  mockdata.horizontal;
const horizontalOrientation = "horizontal";
const verticalOrientation = "vertical";

describe("VerticalTable Component", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <Table
        orientation={verticalOrientation}
        header={verticalHeaders}
        content={verticalContent}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("does not throw error when content is empty", () => {
    const { container } = render(
      <Table
        orientation={verticalOrientation}
        header={verticalHeaders}
        content={[]}
      />,
    );
    expect(container.firstChild).toBeEmptyDOMElement();
  });
});

describe("HorizontalTable Component", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <Table
        orientation={horizontalOrientation}
        header={horizontalHeaders}
        content={horizontalContent}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("header and content included in table when both props are passed", () => {
    const { getByText, getAllByText } = render(
      <Table
        orientation={horizontalOrientation}
        header={horizontalHeaders}
        content={horizontalContent}
      />,
    );
    expect(getByText("Variable value")).toBeInTheDocument();
    expect(getAllByText("multi-instance-process")[0]).toBeInTheDocument();
    expect(getAllByText("multi-instance-process").length).toEqual(6);
  });

  it("renders header, empty row and pagination when the content is empty", () => {
    const component = render(
      <Table
        orientation={horizontalOrientation}
        header={horizontalHeaders}
        content={[]}
      />,
    );
    expect(component.getByRole("table")).toBeInTheDocument();
    expect(component.getAllByRole("row").length).toEqual(3);
    expect(component.getAllByRole("button")[0]).toBeInTheDocument();
  });

  it("does not throw error when all props are empty", () => {
    const { container } = render(
      <Table orientation={horizontalOrientation} header={[]} content={[]} />,
    );
    expect(container).toBeEmptyDOMElement();
  });

  it("table should sort by column ascending when clicking the column name", () => {
    const { getByText } = render(
      <Table
        orientation={horizontalOrientation}
        header={horizontalHeaders}
        content={horizontalContent}
      />,
    );
    fireEvent.click(getByText("Time created"));
    expect(getByText("Time created")).toBeInTheDocument();
    expect(getByText("▲")).toBeInTheDocument();

    const html = document.body.innerHTML;
    const later = html.search("1696667333");
    const earlier = html.search("1696667542");
    expect(later).toBeLessThan(earlier);
  });

  it("table should sort by column descending when clicking the column name twice", () => {
    const { getByText } = render(
      <Table
        orientation={horizontalOrientation}
        header={horizontalHeaders}
        content={horizontalContent}
      />,
    );
    fireEvent.click(getByText("Time created"));
    fireEvent.click(getByText("Time created"));
    expect(getByText("Time created")).toBeInTheDocument();
    expect(getByText("▼")).toBeInTheDocument();

    const html = document.body.innerHTML;
    const later = html.search("1696667333");
    const earlier = html.search("1696667542");
    expect(earlier).toBeLessThan(later);
  });

  it("should set page correctly", () => {
    const { getByText, getByLabelText } = render(
      <Table
        orientation={horizontalOrientation}
        header={horizontalHeaders}
        content={horizontalContent}
      />,
    );

    expect(getByText("1–10 of 12")).toBeInTheDocument();
    fireEvent.click(getByLabelText("Go to next page"));
    expect(getByText("11–12 of 12")).toBeInTheDocument();
  });

  it("should set rows per page correctly", () => {
    const { getByLabelText, container } = render(
      <Table
        orientation={horizontalOrientation}
        header={horizontalHeaders}
        content={horizontalContent}
      />,
    );

    expect(getByLabelText("rows per page")).toHaveValue("10");
    expect(container.querySelector("tbody")?.childElementCount).toEqual(10);

    fireEvent.change(getByLabelText("rows per page"), {
      target: { value: "20" },
    });
    expect(getByLabelText("rows per page")).toHaveValue("20");
    expect(container.querySelector("tbody")?.childElementCount).toEqual(12);
  });
});
