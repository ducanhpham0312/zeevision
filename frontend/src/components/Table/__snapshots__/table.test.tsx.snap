// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalTable Component renders correctly 1`] = `
<DocumentFragment>
  <table
    class="w-full border-collapse rounded bg-white"
  >
    <thead
      class="border-b-2 border-accent font-bold text-text"
    >
      <tr>
        <th
          class="cursor-pointer py-1 text-left"
        >
          <button
            class="group hover:shadow-lg transition relative box-border text-accent hover:bg-hover active:bg-active rounded px-3 py-2 w-full text-left"
          >
            <div
              class="transition-width absolute bottom-0 left-0 w-full transform rounded bg-accent duration-100"
              style="height: 0px;"
            />
            <span>
              <div
                class="relative flex justify-between pr-5"
              >
                <p>
                  Element ID
                </p>
                <div
                  class="absolute right-0"
                />
              </div>
            </span>
          </button>
        </th>
        <th
          class="cursor-pointer py-1 text-left"
        >
          <button
            class="group hover:shadow-lg transition relative box-border text-accent hover:bg-hover active:bg-active rounded px-3 py-2 w-full text-left"
          >
            <div
              class="transition-width absolute bottom-0 left-0 w-full transform rounded bg-accent duration-100"
              style="height: 0px;"
            />
            <span>
              <div
                class="relative flex justify-between pr-5"
              >
                <p>
                  Variable name
                </p>
                <div
                  class="absolute right-0"
                />
              </div>
            </span>
          </button>
        </th>
        <th
          class="cursor-pointer py-1 text-left"
        >
          <button
            class="group hover:shadow-lg transition relative box-border text-accent hover:bg-hover active:bg-active rounded px-3 py-2 w-full text-left"
          >
            <div
              class="transition-width absolute bottom-0 left-0 w-full transform rounded bg-accent duration-100"
              style="height: 0px;"
            />
            <span>
              <div
                class="relative flex justify-between pr-5"
              >
                <p>
                  Variable Value
                </p>
                <div
                  class="absolute right-0"
                />
              </div>
            </span>
          </button>
        </th>
        <th
          class="cursor-pointer py-1 text-left"
        >
          <button
            class="group hover:shadow-lg transition relative box-border text-accent hover:bg-hover active:bg-active rounded px-3 py-2 w-full text-left"
          >
            <div
              class="transition-width absolute bottom-0 left-0 w-full transform rounded bg-accent duration-100"
              style="height: 0px;"
            />
            <span>
              <div
                class="relative flex justify-between pr-5"
              >
                <p>
                  Flow Scope Key
                </p>
                <div
                  class="absolute right-0"
                />
              </div>
            </span>
          </button>
        </th>
        <th
          class="cursor-pointer py-1 text-left"
        >
          <button
            class="group hover:shadow-lg transition relative box-border text-accent hover:bg-hover active:bg-active rounded px-3 py-2 w-full text-left"
          >
            <div
              class="transition-width absolute bottom-0 left-0 w-full transform rounded bg-accent duration-100"
              style="height: 0px;"
            />
            <span>
              <div
                class="relative flex justify-between pr-5"
              >
                <p>
                  Time created
                </p>
                <div
                  class="absolute right-0"
                />
              </div>
            </span>
          </button>
        </th>
        <th
          class="cursor-pointer py-1 text-left"
        >
          <button
            class="group hover:shadow-lg transition relative box-border text-accent hover:bg-hover active:bg-active rounded px-3 py-2 w-full text-left"
          >
            <div
              class="transition-width absolute bottom-0 left-0 w-full transform rounded bg-accent duration-100"
              style="height: 0px;"
            />
            <span>
              <div
                class="relative flex justify-between pr-5"
              >
                <p>
                  Option
                </p>
                <div
                  class="absolute right-0"
                />
              </div>
            </span>
          </button>
        </th>
      </tr>
    </thead>
    <tbody
      aria-label="custom pagination table"
    >
      <tr
        class="border-b border-black/10 bg-second-accent hover:bg-second-accent/20"
      >
        <td
          class="p-3"
        >
          multi-instance-process
        </td>
        <td
          class="p-3"
        >
          items
        </td>
        <td
          class="p-3"
        >
          [{"id":1,"value":10,"inCart":10,"inStock":20},{"id":2,"value":20,"inCart":2,"inStock":10}]
        </td>
        <td
          class="p-3"
        >
          2251799813984108
        </td>
        <td
          class="p-3"
        >
          1696667765
        </td>
        <td
          class="p-3"
        >
          Show History
        </td>
      </tr>
      <tr
        class="border-b border-black/10 hover:bg-second-accent/10"
      >
        <td
          class="p-3"
        >
          multi-instance-process
        </td>
        <td
          class="p-3"
        >
          isValid
        </td>
        <td
          class="p-3"
        >
          true
        </td>
        <td
          class="p-3"
        >
          2251799813984108
        </td>
        <td
          class="p-3"
        >
          1696667765
        </td>
        <td
          class="p-3"
        >
          Show History
        </td>
      </tr>
      <tr
        class="border-b border-black/10 bg-second-accent hover:bg-second-accent/20"
      >
        <td
          class="p-3"
        >
          multi-instance-process
        </td>
        <td
          class="p-3"
        >
          total
        </td>
        <td
          class="p-3"
        >
          200
        </td>
        <td
          class="p-3"
        >
          2251799813984108
        </td>
        <td
          class="p-3"
        >
          1696667765
        </td>
        <td
          class="p-3"
        >
          Show History
        </td>
      </tr>
      <tr
        class="border-b border-black/10 hover:bg-second-accent/10"
      >
        <td
          class="p-3"
        >
          another-multi-instance-process
        </td>
        <td
          class="p-3"
        >
          anotherVariable
        </td>
        <td
          class="p-3"
        >
          string
        </td>
        <td
          class="p-3"
        >
          2251799813984432
        </td>
        <td
          class="p-3"
        >
          1696667333
        </td>
        <td
          class="p-3"
        >
          Show History
        </td>
      </tr>
      <tr
        class="border-b border-black/10 bg-second-accent hover:bg-second-accent/20"
      >
        <td
          class="p-3"
        >
          new-multi-instance-process
        </td>
        <td
          class="p-3"
        >
          kmean
        </td>
        <td
          class="p-3"
        >
          newValue
        </td>
        <td
          class="p-3"
        >
          2251799813984100
        </td>
        <td
          class="p-3"
        >
          1696667542
        </td>
        <td
          class="p-3"
        >
          Show History
        </td>
      </tr>
      <tr
        class="border-b border-black/10 hover:bg-second-accent/10"
      >
        <td
          class="p-3"
        >
          sixth-multi-instance-process
        </td>
        <td
          class="p-3"
        >
          amp
        </td>
        <td
          class="p-3"
        >
          integer
        </td>
        <td
          class="p-3"
        >
          2251799813984642
        </td>
        <td
          class="p-3"
        >
          1696667164
        </td>
        <td
          class="p-3"
        >
          Show History
        </td>
      </tr>
      <tr
        class="border-b border-black/10 bg-second-accent hover:bg-second-accent/20"
      >
        <td
          class="p-3"
        >
          multi-instance-process
        </td>
        <td
          class="p-3"
        >
          items
        </td>
        <td
          class="p-3"
        >
          [{"id":1,"value":10,"inCart":10,"inStock":20},{"id":2,"value":20,"inCart":2,"inStock":10}]
        </td>
        <td
          class="p-3"
        >
          2251799813984108
        </td>
        <td
          class="p-3"
        >
          1696667765
        </td>
        <td
          class="p-3"
        >
          Show History
        </td>
      </tr>
      <tr
        class="border-b border-black/10 hover:bg-second-accent/10"
      >
        <td
          class="p-3"
        >
          multi-instance-process
        </td>
        <td
          class="p-3"
        >
          isValid
        </td>
        <td
          class="p-3"
        >
          true
        </td>
        <td
          class="p-3"
        >
          2251799813984108
        </td>
        <td
          class="p-3"
        >
          1696667765
        </td>
        <td
          class="p-3"
        >
          Show History
        </td>
      </tr>
      <tr
        class="border-b border-black/10 bg-second-accent hover:bg-second-accent/20"
      >
        <td
          class="p-3"
        >
          multi-instance-process
        </td>
        <td
          class="p-3"
        >
          total
        </td>
        <td
          class="p-3"
        >
          200
        </td>
        <td
          class="p-3"
        >
          2251799813984108
        </td>
        <td
          class="p-3"
        >
          1696667765
        </td>
        <td
          class="p-3"
        >
          Show History
        </td>
      </tr>
      <tr
        class="border-b border-black/10 hover:bg-second-accent/10"
      >
        <td
          class="p-3"
        >
          another-multi-instance-process
        </td>
        <td
          class="p-3"
        >
          anotherVariable
        </td>
        <td
          class="p-3"
        >
          string
        </td>
        <td
          class="p-3"
        >
          2251799813984432
        </td>
        <td
          class="p-3"
        >
          1696667333
        </td>
        <td
          class="p-3"
        >
          Show History
        </td>
      </tr>
    </tbody>
    <tfoot>
      <tr>
        <td
          class="MuiTablePagination-root css-7ugj6v"
          colspan="6"
        >
          <div
            class="MuiTablePagination-toolbar"
          >
            <div
              class="MuiTablePagination-spacer"
            />
            <p
              class="MuiTablePagination-selectLabel"
              id=":r1:"
            >
              Rows per page:
            </p>
            <select
              aria-label="rows per page"
              aria-labelledby=":r1: :r0:"
              class="MuiTablePagination-select"
              id=":r0:"
            >
              <option
                class="MuiTablePagination-menuItem"
                value="5"
              >
                5
              </option>
              <option
                class="MuiTablePagination-menuItem"
                value="10"
              >
                10
              </option>
              <option
                class="MuiTablePagination-menuItem"
                value="15"
              >
                15
              </option>
              <option
                class="MuiTablePagination-menuItem"
                value="20"
              >
                20
              </option>
              <option
                class="MuiTablePagination-menuItem"
                value="25"
              >
                25
              </option>
              <option
                class="MuiTablePagination-menuItem"
                value="30"
              >
                30
              </option>
              <option
                class="MuiTablePagination-menuItem"
                value="-1"
              >
                All
              </option>
            </select>
            <p
              class="MuiTablePagination-displayedRows"
            >
              1–10 of 12
            </p>
            <div
              class="MuiTablePagination-actions"
            >
              <button
                aria-label="Go to first page"
                disabled=""
                title="Go to first page"
              >
                <span>
                  |⇽
                </span>
              </button>
              <button
                aria-label="Go to previous page"
                disabled=""
                title="Go to previous page"
              >
                <span>
                  ⇽
                </span>
              </button>
              <button
                aria-label="Go to next page"
                title="Go to next page"
              >
                <span>
                  ⇾
                </span>
              </button>
              <button
                aria-label="Go to last page"
                title="Go to last page"
              >
                <span>
                  ⇾|
                </span>
              </button>
            </div>
          </div>
        </td>
      </tr>
    </tfoot>
  </table>
</DocumentFragment>
`;

exports[`VerticalTable Component renders correctly 1`] = `
<DocumentFragment>
  <table
    class="w-full border-collapse rounded bg-white"
  >
    <tbody>
      <tr
        class="bg-second-accent hover:bg-second-accent/20"
      >
        <th
          class="border border-r-0 p-3 text-left"
        >
          Instance key
        </th>
        <td
          class="border border-l-0 p-3 text-left"
        >
          2251799813984108
        </td>
      </tr>
      <tr
        class="hover:bg-second-accent/10"
      >
        <th
          class="border border-r-0 p-3 text-left"
        >
          BPMN process ID
        </th>
        <td
          class="border border-l-0 p-3 text-left"
        >
          multi-instance-process
        </td>
      </tr>
      <tr
        class="bg-second-accent hover:bg-second-accent/20"
      >
        <th
          class="border border-r-0 p-3 text-left"
        >
          Version
        </th>
        <td
          class="border border-l-0 p-3 text-left"
        >
          1
        </td>
      </tr>
      <tr
        class="hover:bg-second-accent/10"
      >
        <th
          class="border border-r-0 p-3 text-left"
        >
          Process key
        </th>
        <td
          class="border border-l-0 p-3 text-left"
        >
          2251799813685249
        </td>
      </tr>
      <tr
        class="bg-second-accent hover:bg-second-accent/20"
      >
        <th
          class="border border-r-0 p-3 text-left"
        >
          State
        </th>
        <td
          class="border border-l-0 p-3 text-left"
        >
          ELEMENT_ACTIVATING
        </td>
      </tr>
      <tr
        class="hover:bg-second-accent/10"
      >
        <th
          class="border border-r-0 p-3 text-left"
        >
          Start time
        </th>
        <td
          class="border border-l-0 p-3 text-left"
        >
          1696667765
        </td>
      </tr>
      <tr
        class="bg-second-accent hover:bg-second-accent/20"
      >
        <th
          class="border border-r-0 p-3 text-left"
        >
          End time
        </th>
        <td
          class="border border-l-0 p-3 text-left"
        >
          1696667800
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
